lib/ueberauth/strategy/slack.ex

```diff
@@ -133,23 +134,32 @@ defmodule Ueberauth.Strategy.Slack do
     user = conn.private[:slack_user]
     auth = conn.private[:slack_auth]
     identity = conn.private[:slack_identity]
-    image_urls = (get_in(user, ["profile"]) || %{})
+    profile = (get_in(user, ["profile"]) || get_in(identity, ["user"]) || %{})
+    image_urls = profile
     |> Map.keys
     |> Enum.filter(&(&1 =~ ~r/^image_/))
-    |> Enum.map(&({&1, user["profile"][&1]}))
+    |> Enum.map(&({&1, profile[&1]}))
```

Comment:
I know this code was there before your pr, but would it make sense instead of filtering on the keys of the map, filter on the full map, to avoid the map lookup. I am thinking something like this
```
profile
|> Enum.filter(fn {key, value} -> (key =~ ~r/^image_/) end
```